### ISSUE TITLE
`w` in textmode works like `e` in vim

### ISSUE THREAD

**Issue by robrechtdr** [vimperator/vimperator-labs#144]
Imagine I have the following text with my cursor on the first character: `hi lady`.

In vim, when I do `w` the cursor is now over the l. Then I go to insert mode and write `my` I end up with the following: `hi mylady`.

In vimperator when I do the same sequence of commands for the same text in a text field using textarea mode, I end up with the following: `himy lady`. In other words, `w` in textmode currently works like `e` in vim.

**Comment by gkatsev** [comment]
Yeah, looks like `e` and `w` are mapped to the [same](https://github.com/vimperator/vimperator-labs/blob/e6ca25dcb9e789246a5bf10ac1683ebaf330a46e/common/content/editor.js#L154-L157) [thing](https://github.com/vimperator/vimperator-labs/blob/e6ca25dcb9e789246a5bf10ac1683ebaf330a46e/common/content/editor.js#L601).

**Comment by sukima** [comment]
I looked into the [`doCommand`](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XUL/Method/doCommand) which executes the `cmd_selectWordNext`. These are [XUL Commands][1] provided by FireFox (and other mozilla based products). Looking through the [list][1] and there doesn't appear to be a `cmd_selectWordNextEnd` available.

This would explain why it is better that <kbd>e</kbd> does the same as <kbd>w</kbd> so that at least pressing the key doesn't do nothing even if it's not quite the right thing.

A possible workaround might be to execute two commands: go to next word then move cursor to the left one. But his would be problematic for situations where there were more then one character between words of possibly if the cursor can't to next word as it's end of line then cursor moves left unexpectedly. It's feels like a rat's nest to workaround the lack of such an option in the XUL API.

Thoughts / Comments?

[1]: https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XUL/List_of_commands

**Comment by maxauthority** [comment]
Well, you investigated the issue correctly, but as you see there is no
simple solution. The right thing would be to implement the logic oneself
but that's certainly a major task. You might take a look in how the
TEXTAREA mode is handled when you go into "vim mode" of a textarea with
ctrl-t. Maybe this does the right thing, but I doubt it.

**Comment by sukima** [comment]
Same code, same logic, same problem in both caret and textarea modes. Bummer.

**Comment by jamalis** [comment]
Gee thanks a lot, with the post above I became rather well-informed about the discussions.
http://fotorumahmewah.com
